### PR TITLE
Object counting by areas

### LINKED ISSUE #355: Object counting by areas
We need a simple sample demonstrating people counting for defined areas. An area is defined by a polygon, name, and color. The demo involves a sidebar legend with a name, color, and an amount of counted objects within it.

Use a traffic_meter as a base implementation. 

The name is `area_object_counting`.

### PR DESCRIPTION
**Pull request by abramov-oleg**

### CODE REVIEW

File: samples/area_object_counting/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+The pipeline determines if people are located in the delimited areas on the frame and counts the amount of people for each area displaying the counts on the sidebar.
+
+Preview:
+
+![](assets/area-object-counting-loop.webp)
+
+Tested on platforms:
+
+- Xavier NX, Xavier AGX;
```
Comment (dorgun) [inline samples/area_object_counting/README.md:11]: Was it tested on the Orin Nano?

File: savant/utils/artist/artist_gpumat.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def add_text(
         bg_color: Optional[Tuple[int, int, int, int]] = (0, 0, 0, 255),  # black
         padding: Tuple[int, int, int, int] = (0, 0, 0, 0),
         anchor_point_type: Position = Position.CENTER,
-    ) -> int:
+    ) -> Tuple[Tuple[int, int], int]:
```
Comment (dorgun) [inline savant/utils/artist/artist_gpumat.py:55]: We have a release with non-breaking changes. It's certainly not a critical change, but if someone uses the return value of the add_text method, users will get an error. Is it possible to draw without these changes?

Comment (bwsw) [reply savant/utils/artist/artist_gpumat.py:55]: @dorgun @abramov-oleg I think it is ok.

Comment (abramov-oleg) [reply savant/utils/artist/artist_gpumat.py:55]: Centering draw objects (such as boxes) vertically against text characters (and not text entire text box) can be challenging without this change.

File: samples/area_object_counting/README.md (hunk 1 above)
Comment (dorgun) [reply samples/area_object_counting/README.md:11]: Orin support will be added after #527

**Review by dorgun — APPROVED** [review]

**Pull request merged**